app/controllers/miq_ae_class_controller.rb

```diff
@@ -1822,11 +1825,14 @@ def create_action_url(node)
 
   def get_rec_name(rec)
     column = rec.display_name.blank? ? :name : :display_name
-    if rec.kind_of?(MiqAeNamespace) && rec.domain? && (!rec.editable? || !rec.enabled)
-      add_read_only_suffix(rec, rec.send(column))
-    else
-      rec.send(column)
+    if rec.kind_of?(MiqAeNamespace) && rec.domain?
+      editable_domain = User.current_tenant.editable_domains.include?(rec)
```

Comment:
Honestly, I don't care where we get the results, I don't want to get a list of all editable domains and then see if the current record is in that list.  We have the ability to know if a domain is editable for me so we should just ask that object or a more global `can?` method... What the object or `can?` does under the covers to answer that question is not important to me here.  `can?(:edit, rec)` is fine too, just keep the logic of how to tell if the thing is editable outside of the controller..